types/models.d.ts

```diff
@@ -1,9 +1,9 @@
 declare module 'mongoose' {
   import mongodb = require('mongodb');
 
-  export interface AcceptsDiscriminator {
+  export interface AcceptsDiscriminator<B> {
     /** Adds a discriminator type. */
-    discriminator<D>(name: string | number, schema: Schema, value?: string | number | ObjectId): Model<D>;
+    discriminator<D>(name: string | number, schema: Schema<D>, value?: string | number | ObjectId): Model<B & D>;
```

Comment:
Probably better to use `Omit<B, keyof D> & D` instead of `B & D`. Discriminator schemas can overwrite parent schema properties.

Reply:
This is indeed a bug. Fixed!